dotCMS/src/main/java/com/dotmarketing/db/listeners/CommitAPI.java

```diff
@@ -0,0 +1,349 @@
+package com.dotmarketing.db.listeners;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.google.common.annotations.VisibleForTesting;
+
+public class CommitAPI {
+
+
+    public enum CommitListenerStatus {
+        DISABLED, ALLOW_SYNC, ALLOW_ASYNC;
+    }
+    
+    private static class apiHolder {
+        static final CommitAPI api = new CommitAPI();
+    }
+
+    public static CommitAPI getInstance() {
+        return apiHolder.api;
+    }
+    
+    private CommitListenerStatus nextStatus = CommitListenerStatus.ALLOW_ASYNC;
+    
+    
+    public void forceStatus(final CommitListenerStatus status) {
+        this.nextStatus=status;
+    }
+    
+    /**
+     * Status for listeners of thread-local -based transactions. This allows to control as to which
+     * listeners are fired on a commit
+     */
+    private final ThreadLocal<CommitListenerStatus> listenersStatus = new ThreadLocal<CommitListenerStatus>() {
+        protected CommitListenerStatus initialValue() {
+            return CommitListenerStatus.ALLOW_ASYNC;
+        }
+    };
+
+
+    /**
+     * Returns the listeners status currently associated to the thread-local -based transaction
+     * (ALLOW_ASYNC by default)
+     */
+    public CommitListenerStatus status() {
+        return listenersStatus.get();
+    }
+
+    /**
+     * Allows to override the status of the listeners associated to the current thread-local -based
+     * transaction (DISABLED if overriden) When using TransactionListenerStatus.DISABLED, client code
+     * should be aware of controlling the operations that are suppossed to be done by listeners
+     * 
+     * @param status TransactionListenerStatus
+     */
+    public void setCommitListenerStatus(CommitListenerStatus status) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'status' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5639225)